sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/UnionEstimation.scala

```diff
@@ -70,8 +70,20 @@ object UnionEstimation {
       None
     }
 
-    val unionOutput = union.output
+    val (attrToComputeMinMaxStats, attrToComputeNullCount) = getAttributeToComputeStats(union)
+    val attrStatsWithMinMax = computeMinMaxAttributeStatsMap(union, attrToComputeMinMaxStats)
+    val newAttrStats = updateAttributeStatsMapWithNullCount(union, attrStatsWithMinMax,
+      attrToComputeNullCount)
+
+    Some(
+      Statistics(
+        sizeInBytes = sizeInBytes,
+        rowCount = outputRows,
+        attributeStats = newAttrStats))
+  }
 
+  private def getAttributeToComputeStats(union: Union) = {
```

Comment:
We need this method? I think its okay to check if we need to update stats or not in the `computeMinMaxAttributeStatsMap` and `updateAttributeStatsMapWithNullCount` sides. How about organizing the `estimate` method like this?
```
def estimate(union: Union).... = {
  ...
  val newMinMaxStats = computeNewMinMaxStats(union)
  val newNullCountStats = computeNewNullCountStats(union)

  val newAttrStats = (builds a new stats from `newMinMaxStats` and `newNullCountStats`)
  Some(
    Statistics(...
```

Reply:
Removed the extra method and updated the code.